vignettes/implementing-cookies.Rmd

```diff
@@ -0,0 +1,110 @@
+
+dfeshiny provides a standard GDS compliant banner, a cookie information panel 
+and the underlying code required to allow end users to give or withhold their 
+consent for cookies to be used. If they withhold their consent, then tracking 
+with Google Analytics will be halted.
+
+Implementing cookie control on your dashboard is handled by several functions 
+in dfeshiny, with the process outlined below.
+
+The cookie control requires some underlying javascript to remove and create 
+cookies as well as switch Google Analytics on and off. This javascript needs 
+including in your dashboard's www/ folder, which can be achieved by running the 
+following command from the R console in your dashboard project / repo:
+
+```{r eval=FALSE}
+dfeshiny::init_cookies()
+```
+
+Before adding the cookie consent banner, we recommend adding your Google 
+Analytics key as a variable to your dashboard's **[global.R](https://github.com/dfe-analytical-services/dfeshiny/blob/cookie-module/tests/test_dashboard/global.R)**
+file (replacing `ABCDE12345` below with the key provided by the Statistics 
+Development team):
+
+```{r eval=FALSE}
+google_analytics_key <- "ABCDE12345"
+```
+
+You will then need to add the necessary code to your **ui.R** file to a) run the 
+javascript (`dfe_cookie_script`), b) create the consent banner 
+(`cookie_banner_ui`) and c) add a cookies information panel 
+(`cookies_panel_ui`):
+
+
+```{r eval=FALSE}
+ui <- function(input, output, session) {
+  shiny::fluidPage(
+    shinyjs::useShinyjs(),
+    dfeshiny::dfe_cookie_script(),
+    dfeshiny::cookie_banner_ui(
+      "cookie-banner",
+      "Your dashboard name"
+    ),
+    shiny::navlistPanel(
+      "",
+      id = "navlistPanel",
+      widths = c(2, 8),
+      well = FALSE,
+      dfeshiny::cookies_panel_ui(
+        id = "cookie-panel",
+        google_analytics_key = google_analytics_key
+      )
+    )
+  )
+}
+```
+
+You should then add the `cookie_banner_server` and `cookies_panel_server` 
+functions to your **server.R** script:
+
+```{r eval=FALSE}
+output$cookie_status <- dfeshiny::cookie_banner_server(
+  "cookie-banner",
+  input_cookies = shiny::reactive(input$cookies),
+  parent_session = session,
+  google_analytics_key = google_analytics_key,
+  cookie_link_panel = "cookies_panel_ui"
+)
+
+dfeshiny::cookies_panel_server(
+  id = "cookie-panel",
+  input_cookies = shiny::reactive(input$cookies),
+  google_analytics_key = google_analytics_key
+)
+```
+
+In the above you should keep all of the flags as they are shown. Note that this 
+assumes you have added `google_analytics_key` as a declared variable in your 
+global.R file. If you've not done this, you'll need to replace 
+`google_analytics_key = google_analytics_key` to 
+`google_analytics_key = "ABCDE12345"` (replacing "ABCDE12345" with your own 
+key).
+
+With the above included, when the user's preference will itself be stored in a
```

Comment:
I think the "when" in this sentence could go!

Reply:
Removed.